alf/layers.py

```diff
@@ -705,6 +706,137 @@ def bias(self):
         return self._bias
 
 
+@alf.configurable
+class CausalConv1D(nn.Module):
+    """1D (Dilated) Causal Convolution layer.
+        1D Dilated Causal Convolution is proposed in `Aaron et. al. WaveNet:
+        A generative model for raw audio <https://arxiv.org/abs/1609.03499>`_
+    """
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 kernel_size,
+                 dilation=1,
+                 hide_current=False,
+                 activation=torch.relu_,
+                 use_bias=None,
+                 use_bn=False,
+                 kernel_initializer=None,
+                 kernel_init_gain=1.0,
+                 bias_init_value=0.0):
+        """A layer implementing the 1D (Dilated) Causal Convolution.
+        It is also responsible for activation and customized weights
+        initialization. An auto gain calculation might depend on the activation
+        following the causal conv1d layer.
+
+        Note that the main difference of causal conv v.s. standard conv is that
+        each temporal element in the convolutional output is causal w.r.t.
+        the temporal elements from input. For example, for a length L sequence
+        x with the shape of [B, C, L], and y = causal_conv(x), where the shape
+        of y is [B, C', L], by causal we mean y[..., l] only depends on
+        X[..., :l] (i.e. the past), and there is no dependency on X[..., :l]
+        (i.e. future) as in the standard non-causal convolution.
```

Comment:
Use correct formatting for these math notations.

Reply:
Fixed